mlir/include/mlir/IR/AffineExpr.h

```diff
@@ -96,6 +96,20 @@ class AffineExpr {
   /// constants, i.e., it does not involve dimensional identifiers.
   bool isSymbolicOrConstant() const;
 
+  /// Returns true if this expression is made out of only symbols.
+  bool isSymbolic() const;
+
+  /// Returns true if the expression is divisible by the given symbol.
+  bool isSymbolicDivisible() const;
+
+  /// Returns a affine expression by dividing given affine expression by given
+  /// symbol
+  AffineExpr symbolicDivide();
+
+  /// Simplify an semi-affine expression by handling modulo, floordiv or ceildiv
+  /// by any symbol.
+  AffineExpr simplifySemi();
```

Comment:
Adding these as class methods is way out of line. Instead, all of these should go into the existing expression simplification methods.

Reply:
Because of its difference from other simplification methods, it is a separate function. It might be possible name is suggesting totally different function for simplification but what it is doing is assisting the simplifyaffineexpr function only.